docs/architecture/api-packages.md

```diff
@@ -0,0 +1,156 @@
+# API Package
+
+## Introduction
+
+Previously, in Kyma Runtime, single API or Event Definition of a given Application was represented as a single Service Class in Service Catalog.
+
+To have control over this behavior, we introduce API Packages. API Package is a way of grouping multiple API or Event Definitions, so that they are represented as single Service Class.
+
+## Solution
+
+![API Packages Diagram](./assets/api-packages.svg)
+
+In order to create API packages, Compass utilize Application labels. On Application level, one can specify package by using label in a form of the following key and value:
+
+**Key:** `api-packages`
+**Value:**
+
+```json
+{
+  "foo": {
+    "description": "",
+    "metadata": {
+      "displayName": "",
+      "documentationURL": "",
+      "supportURL": ""
+    },
+    "events": [
+      "707ea9da-0bf8-4dad-8145-a934ee1ebffb",
+      "6e5969c6-63fd-42c0-af17-72ed95c9dcd5"
+    ],
+    "apis": ["foo", "bar"]
+  }
+}
+```
+
+The root object key defines API Package name. In this example, `foo` is the API Package name.
+
+In order to allow validation of the label value, Compass creates `api-packages` LabelDefinition automatically. It is done while creating new Application. Once Tenant management feature is ready, creating new tenant will be used as a trigger for the LabelDefinition creation.
+The `api-packages` LabelDefinition has the following JSON schema:
+
+```json
+{
+  "$id": "https://example.com/arrays.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "A representation of a person, company, organization, or place",
+  "type": "object",
+  "propertyNames": {
+    "pattern": "^[a-z0-9_-]*$",
+    "minLength": 3,
+    "maxLength": 35
+  },
+  "additionalProperties": {
+    "anyOf": [{ "$ref": "#/definitions/apiPackage" }]
+  },
+  "definitions": {
+    "apiPackage": {
+      "type": "object",
+      "required": ["events", "apis"],
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Description"
+        },
+        "metadata": {
+          "type": "object",
+          "description": "Additional Metadata for Service Class representation"
+        },
+        "events": {
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "pattern": "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+In a result, the `api-packages` label value is validated. Also, the API Package name uniqueness within Application is guaranteed. Event and API Definition existence is not checked, apart from verifying the UUID type of the ID arrays.
+
+### Assumptions
+
+- API Package name doesn't have to be unique within tenant.
```

Comment:
I miss here the fact that we limit the usage of the API Package to the Application scope. In this concept, you cannot mix under one API Package APIs or Events from different applications.

Reply:
Yes, this is also the assumption. I will update the paragraph